elf/elf.go

```diff
@@ -442,6 +451,18 @@ func (b *Module) relocate(data []byte, rdata []byte) error {
 type SectionParams struct {
 	PerfRingBufferPageCount   int
 	SkipPerfMapInitialization bool
+	CustomConfig              []PinCustomConfig
+}
+
+// PinCustomConfig represents a custom config for pinning.
+// Id has to be greater than 2, as numbers 0,1,2 are reserved.
+// The final path will be composed in a form of "/sys/fs/bpf/PinPath",
+// where "/sys/fs/bpf" is a mount point of the standard bpf filesystem.
+// PinPath can be any path, e.g. a normal path like "ns1/testgroup1",
+// or sha256 string of a particular path.
+type PinCustomConfig struct {
+	Id      uint32 // >= 3
```

Comment:
`Id` doesn't seem to have any actual purpose? Do I miss something?

Reply:
It has to match the `pinning` value in `struct bpf_map_def`:
https://github.com/iovisor/gobpf/blob/09e8636/elf/elf.go#L68

One bpf module can have several maps, and each map can have a different `pinning` field. So we need to be able to pass several `PinCustomConfig` configuration in the `Load()` to define potentially several mappings.

Reply:
> It has to match the `pinning` value in struct `bpf_map_def`

Hmm, I don't see the purpose,  seem redundant in the case of maps and not needed here, since the PR doesn't consider custom pinning for maps during `Load()`, if I haven't missed something.

> One bpf module can have several maps, and each map can have a different pinning field.

But we can have only one map per section, and the parameters are per section (`SectionParams`)?

Reply:
> But we can have only one map per section, and the parameters are per section (`SectionParams`)?

Oh, I missed that...